.changeset/slimy-mugs-type.md

```diff
@@ -0,0 +1,5 @@
+---
+"@firebase/firestore": patch
+---
+
+Fixes an issue in the Transaction API that caused the SDK to return invalid DocumentReferences when returned from `DocumentSnapshot.data()`.
```

Comment:
How about "caused `DocumentSnapshot.data()` to return valid DocumentReferences"? The mixing of present and past tense for "return" makes it a bit hard to understand.

Reply:
I changed it up a bit.